com.unity.render-pipelines.high-definition/Editor/Lighting/LightUnit/LightUnitSliderSettings.cs

```diff
@@ -0,0 +1,208 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.Rendering.HighDefinition;
+
+namespace UnityEditor.Rendering.HighDefinition
+{
+    struct LightUnitSliderUIDescriptor
+    {
+        public LightUnitSliderUIDescriptor(LightUnitSliderUIRange[] valueRanges, string cautionTooltip, string unitName, bool hasMarkers = true)
+        {
+            this.valueRanges = valueRanges;
+            this.cautionTooltip = cautionTooltip;
+            this.unitName = unitName;
+            this.hasMarkers = hasMarkers;
+
+            sliderRange = new Vector2(
+                this.valueRanges.Min(x => x.value.x),
+                this.valueRanges.Max(x => x.value.y)
+            );
+        }
+
+        public readonly LightUnitSliderUIRange[] valueRanges;
+        public readonly Vector2 sliderRange;
+        public readonly string cautionTooltip;
+        public readonly string unitName;
+        public readonly bool hasMarkers;
+    }
+
+    struct LightUnitSliderUIRange
+    {
+        public LightUnitSliderUIRange(Texture2D icon, string tooltip, Vector2 value)
+        {
+            this.content = new GUIContent(icon, tooltip);
+            this.value = value;
+        }
+
+        public static LightUnitSliderUIRange CautionRange(string tooltip, float value) => new LightUnitSliderUIRange
+        {
+            // Load the buildin caution icon with provided tooltip.
+            content = new GUIContent( EditorGUIUtility.IconContent("console.warnicon.sml").image, tooltip),
+            value = new Vector2(-1, value)
+        };
+
+        public GUIContent content;
+        public Vector2    value;
+    }
+
+    static class LightUnitSliderDescriptors
+    {
+        // Lux
+        public static LightUnitSliderUIDescriptor k_LuxDescriptor = new LightUnitSliderUIDescriptor(
```

Comment:
`public static` don't follow the `k_` standard, you can directly use pascal case here and for other instances.